### PR TITLE
[508] Date widget a11y updates

### PR DESCRIPTION
**Pull request by Mottie**
## Description

To improve accessibility with the form system `DateWidget`, some modifications have been included in this PR:

- Centralize date min & max year range (1900 to current year + 100) 
- Show date out-of-range error
- Fully spell out the month because screenreaders inconsistently read out the month abbreviations
- Add changes to form 526 military history page

Related:
- Out of range date: https://github.com/department-of-veterans-affairs/va.gov-team/issues/18578 + https://github.com/department-of-veterans-affairs/component-library/pull/26
- Month spelled out fully: https://github.com/department-of-veterans-affairs/component-library/pull/16#discussion_r568320847
- Adding `usa-datefields` class: https://github.com/department-of-veterans-affairs/component-library/pull/16#discussion_r562242079

## Testing done

Updated unit tests

## Screenshots

<details><summary>GIF showing 3 error messages for 526</summary>

year</details>

## Acceptance criteria
- [x] Centralize min/max year constants
- [x] Show date out-of-range error
- [x] Fully spell out month name

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/widgets/DateWidget.jsx [hunk 1]
```diff
@@ -91,12 +98,13 @@ export default class DateWidget extends React.Component {
             id={`${id}Month`}
             value={month}
             disabled={disabled}
+            aria-describedby={`${id}-label ${id}-${'error-message'}`}
```
Comment (Mottie) [inline src/platform/forms-system/src/js/widgets/DateWidget.jsx:101]: I've included both the label (a `legend` in this case) and the error span - I'm not sure if including the ID here if it doesn't exist is a problem for all screenreaders.

**Review by joshkimux — APPROVED** [review]
@Mottie what would we do without you! Thanks for these changes!! 🥳

File: src/platform/forms-system/src/js/widgets/DateWidget.jsx (hunk 1 above)
Comment (1Copenut) [reply src/platform/forms-system/src/js/widgets/DateWidget.jsx:101]: I'm on the fence about adding the `aria-describedby` here. The legend is read aloud the first time users enter the group if they're moving by `TAB` keypress, and is read inline if users are moving by text node. The attribute could also throw an axe error if it's present and the ID attribute isn't.

All that to say, I think we'll be better off removing the `aria-describedby` attribute.

File: src/platform/forms-system/src/js/utilities/date/index.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-  { label: 'Apr', value: 4 },
-  { label: 'May', value: 5 },
-  { label: 'Jun', value: 6 },
-  { label: 'Jul', value: 7 },
-  { label: 'Aug', value: 8 },
-  { label: 'Sep', value: 9 },
-  { label: 'Oct', value: 10 },
-  { label: 'Nov', value: 11 },
-  { label: 'Dec', value: 12 },
+  { label: 'Jan', value: 1, text: 'January' },
```
Comment (1Copenut) [inline src/platform/forms-system/src/js/utilities/date/index.js:4]: These are good improvements. Some of the longest labels are getting cut off when the select is closed. I'll add a screenshot to https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/394 to capture that new success criteria.

**Review by 1Copenut — CHANGES_REQUESTED** [review]

File: src/platform/forms-system/src/js/widgets/DateWidget.jsx (hunk 1 above)
Comment (Mottie) [reply src/platform/forms-system/src/js/widgets/DateWidget.jsx:101]: It should be okay if the error span ID existed, but the error element content was empty?

Comment (Mottie) [reply src/platform/forms-system/src/js/widgets/DateWidget.jsx]: Hmm, actually the error state and the widget are decoupled, so there isn't an easy way to tell if there is an error. I'll go ahead and remove the `aria-describedby` for now.

File: src/platform/forms-system/src/js/utilities/validations/index.js [hunk 3]
```diff
@@ -32,13 +34,7 @@ export function isValidSSN(value) {
...
 
 export function isValidYear(value) {
-  return (
-    Number(value) >= 1900 &&
-    Number(value) <=
-      moment()
-        .add(100, 'year')
-        .year()
-  );
+  return Number(value) >= minYear && Number(value) <= maxYear;
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/utilities/validations/index.js:37]: I did a double take when I saw this function and the same `isValidYear` in the _other_ validations file :see_no_evil:

File: src/platform/forms-system/src/js/utilities/date/index.js (hunk 2 above)
Comment (bkjohnson) [reply src/platform/forms-system/src/js/utilities/date/index.js:4]: I agree! Looks like we shouldn't have even been abbreviating April, May, June, and July:

https://design.va.gov/content-style-guide/dates-and-numbers#dates-and-years

As a sidenote, do we need the `label` anymore?

**Review by bkjohnson — APPROVED** [review]
:+1:

File: src/platform/forms-system/src/js/utilities/date/index.js (hunk 2 above)
Comment (Mottie) [reply src/platform/forms-system/src/js/utilities/date/index.js:4]: I don't know if it's used by other apps... so I left it.

**Review by 1Copenut — APPROVED** [review]
👍 LGTM! Thank you @Mottie

**Review by ndsprinkle — APPROVED** [review]

**Review by jbalboni — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]

**Review by narin — APPROVED** [review]

File: src/platform/forms-system/src/js/utilities/date/index.js (hunk 2 above)
Comment (stephenBarrs) [reply src/platform/forms-system/src/js/utilities/date/index.js:4]: @Mottie @1Copenut I still see the longer month names being cut off when the select is closed.  Is that planned to be fixed with this PR?
image

Comment (Mottie) [reply src/platform/forms-system/src/js/utilities/date/index.js:4]: Making the Month select wider will require a separate PR to the `veteran-facing-services-tools` repo to modify this entry:

```css
.form-datefield-month {
  width: 12rem;
}
```

Comment (Mottie) [reply src/platform/forms-system/src/js/utilities/date/index.js:4]: See https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/588

**Review by stephenBarrs — APPROVED** [review]

**Pull request merged**